### DISCUSSION TITLE
No lines read from a pbf file

### DISCUSSION THREAD

**Discussion by Zverik** [systemed/tilemaker#789] [Q&A] [answered]
Hi folks. I'm generating a pmtiles with highways for my city, but I don't see any. Tilemaker output is:

```
Layer transportation (z4-14)
Layer transportation_name (z10-14)
Bounding box 24.3917, 59.236, 24.9856, 59.5319
Generated points: 0, lines: 0, polygons: 0
Reading .pbf /data/kiirus.osm.pbf
(3 ms)
Block 13/14 (268 ms)
Block 1/2 (64 ms)
only 0 relation blocks; subdividing for better parallelism
(1 ms)
Generated points: 0, lines: 0, polygons: 0
Attributes: 3830 sets from 26624 objects (9216 uncached), 18432 pairs (7168 uncached)
Creating pmtiles at /data/kiirus.pmtiles
indexed 131 contended objects
osm: finalizing z6 tile 4096/4096 (3 ms)
osm: finalizing z6 tile 4096/4096 (0 ms)
indexed 0 contended objects
shp: finalizing z6 tile 4096/4096 (0 ms)
shp: finalizing z6 tile 4096/4096 (0 ms)
collecting tiles: 0ms, filtering tiles: z8 (1, 0ms) z9 (2, 0ms) z10 (6, 1ms) z11 (20, 6ms) z12 (51, 25ms) z13 (164, 100ms) z14 (569, 406ms)
z6/36/18, writing tile 813 of 813               
Closing pmtiles file
Memory used: 1060540
```

I'm looking at "Generated lines: 0" message. How can it be? The source is a 2 MB pbf, I open it in JOSM and see all the roads.

Running it like this:

```sh
docker run -it --rm --pull always -v $(pwd):/data \
  ghcr.io/systemed/tilemaker:master \
  --verbose \
  /data/kiirus.osm.pbf \
  --output /data/kiirus.pmtiles \
  --config /data/config-kiirus.json \
  --process /data/process-kiirus.lua
```

My config:

```json
{
        "layers": {
                "transportation":      { "minzoom":  8, "maxzoom": 14, "simplify_below": 13, "simplify_level": 0.0003, "simplify_ratio": 2.0 },
                "transportation_name": { "minzoom": 10, "maxzoom": 14 }
        },
        "settings": {
                "minzoom": 8,
                "maxzoom": 14,
                "basezoom": 14,
                "include_ids": false,
                "name": "Kiirusepiirangud",
                "version": "0.1",
                "description": "Speed limit tiles",
    "bounding_box": [24.39168153552042, 59.23603991760673, 24.985629899778232, 59.531880901429716],
                "compress": "gzip"
        }
}
```

And processing script:

```lua
function way_function()
        local highway  = Find("highway")

        -- Roads
        if highway~="" then
                Layer("transportation", false)
                if highway=="unclassified" or highway=="residential" then highway="minor" end
                Attribute("class", highway)
                Attribute("maxspeed", Find("maxspeed"))
                -- ...and road names
                local name = Find("name")
                if name~="" then
                        Layer("transportation_name", false)
                        Attribute("class", highway)
                        Attribute("name", name)
                end
        end
end
```

Sorry if this is a dumb question, I have spent like an hour debugging and googling, and now out of ideas.

**Comment by systemed** [comment]
Can you upload your .pbf somewhere?

**Reply by Zverik** [reply]
Sure: https://osmz.ru/kiirus/kiirus.osm.pbf and my result https://osmz.ru/kiirus/kiirus.pmtiles

**Comment by systemed** [comment, marked as answer]
Hm. If I run that with the default tilemaker profile it happily produces tiles:

    tilemaker --bbox 24.3917,59.236,24.9856,59.5319 ~/Downloads/kiirus.osm.pbf ~/Downloads/kiirus.mbtiles

![Screenshot 2024-12-17 at 15 48 00](https://github.com/user-attachments/assets/64462a19-5e09-4720-98ca-541327f759b8)

That suggests the .pbf is ok, so it's either an issue with the config/Lua, or with whatever you're using to view it. I can't instantly see a problem with the config/Lua though I haven't yet tried them.

I usually suggest using [vt2geojson](https://github.com/mapbox/vt2geojson) to debug vector tiles - spin up your server, then run (say) `vt2geojson http://localhost:8080/13/4658/2405.pbf` (or wherever you're hosting the tiles). This should spit out a load of GeoJSON to stdout with your vector tile contents.

**Comment by systemed** [comment]
Just tried it with the Lua and config and it works fine. So I think it's most probably an issue with whatever you're using to serve it.

Maybe try outputting to .mbtiles rather than .pmtiles, and testing with the bundled server:

    tilemaker-server kiirus.mbtiles --static /path/to/your/tilemaker/server/static/

**Comment by systemed** [comment]
(The other tool you can use to quickly debug an .mbtiles is QGIS - just drag it into your project and it should appear as a new layer)

**Comment by Zverik** [comment]
Okay this was indeed my fault — I only checked the tiles in MapLibre, but it seems the browser has cached something. I've checked the tiles in QGIS, and saw all the data. Sorry Richard, and thank you :)
